experiment/migrate_testgrid_tabs.py

```diff
@@ -0,0 +1,198 @@
+                                  Loader=ruamel.yaml.RoundTripLoader,
+                                  preserve_quotes=True)
+
+    for dashboard in config["dashboards"]:
+        if "dashboard_tab" in dashboard:
+            for dashTab in dashboard["dashboard_tab"][:]:
+                if assert_tab_keys(dashTab):
+                    move_tab(dashboard, dashTab, prow_dir)
+
+    with open(testgrid_config, "w") as confFile:
+        ruamel.yaml.dump(config, confFile, Dumper=ruamel.yaml.RoundTripDumper)
+        confFile.truncate()
+
+
+def move_tab(dashboard, dashTab, prow_dir):
+    """Moves a given dashboard to the matching prow job in prow_dir, if possible"""
+    dashboardName = dashboard["name"]
+    prowJobName = dashTab["test_group_name"]
+    prowJobFileName = find_prow_job(prowJobName, prow_dir)
+    if prowJobFileName == "":
+        return
+
+    # Matching file found; patch and delete
+    print("Patching {0} in {1}".format(prowJobName, prowJobFileName))
+
+    with open(prowJobFileName, "r") as prowJobFile:
+        prowConfig = ruamel.yaml.load(prowJobFile,
+                                      Loader=ruamel.yaml.RoundTripLoader,
+                                      preserve_quotes=True)
+
+    # For each presubmits, postsubmits, periodic:
+    # presubmits -> <any repository> -> [{name: prowjob}]
+    if "presubmits" in prowConfig:
+        for _, jobs in prowConfig["presubmits"].items():
+            for job in jobs:
+                if prowJobName == job["name"]:
+                    job = patch_prow_job(job, dashTab, dashboardName)
+
+    # postsubmits -> <any repository> -> [{name: prowjob}]
+    if "postsubmits" in prowConfig:
+        for _, jobs in prowConfig["postsubmits"].items():
+            for job in jobs:
+                if prowJobName == job["name"]:
+                    job = patch_prow_job(job, dashTab, dashboardName)
+
+    # periodics -> [{name: prowjob}]
+    if "periodics" in prowConfig:
+        for job in prowConfig["periodics"]:
+            if prowJobName == job["name"]:
+                job = patch_prow_job(job, dashTab, dashboardName)
+
+    # Dump ProwConfig to prowJobFile
+    with open(prowJobFileName, "w") as prowJobFile:
+        ruamel.yaml.dump(prowConfig, prowJobFile, Dumper=ruamel.yaml.RoundTripDumper)
+        prowJobFile.truncate()
+
+    # delete tab
+    dashboard["dashboard_tab"].remove(dashTab)
+
+
+def assert_tab_keys(tab):
+    """Asserts if a dashboard tab is able to be migrated.
+
+    To be migratable, the annotations must contain only the following:
+    name, description, test_group_name, alert_options->alert_mail_to_addresses
+    """
+    whitelist = ["name", "description", "test_group_name", "alert_options"]
+    if [k for k in tab.keys() if k not in whitelist]:
+        return False
+
+    if "alert_options" in tab:
+        alertKeys = tab["alert_options"].keys()
+        if len(alertKeys) != 1 or "alert_mail_to_addresses" not in alertKeys:
```

Comment:
nit: could redo this as another `whitelist`-style check so it's easier to add a new key if we find we want to support more `alert_options` keys